shared/pingpong/accounts.go

```diff
@@ -82,186 +85,233 @@ func ensureAccounts(ac libgoal.Client, initCfg PpConfig) (accounts map[string]ui
 
-				tx, err = client.FillUnsignedTxTemplate(addr, 0, 0, cfg.MaxFee, tx)
-				if err != nil {
-					fmt.Printf("Cannot fill asset %v init txn in account %v\n", k, addr)
-					return
-				}
+	// Step 3) Evenly distribute the assets across all participant accounts
+	for addr := range accounts {
+		fmt.Printf("Distributing assets from account %v\n", addr)
+		addrAccount, addrErr := client.AccountInformation(addr)
+		if addrErr != nil {
+			fmt.Printf("Cannot lookup source account\n")
+			err = addrErr
+			return
+		}
 
-				signedTxn, signErr := client.SignTransactionWithWallet(h, nil, tx)
-				if signErr != nil {
-					fmt.Printf("Cannot sign asset %v init txn in account %v\n", k, addr)
-					err = signErr
-					return
-				}
+		assetParams := addrAccount.AssetParams
+		fmt.Printf("Distributing  %d assets\n", len(assetParams))
 
-				_, broadcastErr := client.BroadcastTransaction(signedTxn)
-				if broadcastErr != nil {
-					fmt.Printf("Cannot broadcast asset %v init txn in account %v\n", k, addr)
-					err = broadcastErr
-					return
-				}
+		// Distribute assets to each account
+		for k := range assetParams {
 
-				if !cfg.Quiet {
-					fmt.Printf("Init asset %v in account %v\n", k, addr)
-				}
-				accounts[addr] -= tx.Fee.Raw
-			}
+			fmt.Printf("Distributing asset %v \n", k)
 
-			// fund asset
-			var assetAmt uint64
-			if asset, ok := addrAccount.Assets[k]; ok {
-				if asset.Amount > cfg.MinAccountAsset {
-					assetAmt = 0
-				} else {
-					assetAmt = cfg.MinAccountAsset - asset.Amount
-				}
-			} else {
-				assetAmt = cfg.MinAccountAsset
-			}
+			assetAmt := assetParams[k].Total / uint64(len(accounts))
+			for addr2 := range accounts {
+				if addr != addr2 {
 
-			if assetAmt == 0 {
-				continue
-			}
+					fmt.Printf("Distributing assets from %v to %v \n", addr, addr2)
 
-			tx, sendErr := constructTxn(cfg.SrcAccount, addr, cfg.MaxFee, assetAmt, k, client, cfg)
-			if sendErr != nil {
-				fmt.Printf("Cannot initiate asset %v in account %v\n", k, addr)
-				err = sendErr
-				return
-			}
-			stxn, signErr := signTxn(cfg.SrcAccount, tx, client, cfg)
-			if signErr != nil {
-				fmt.Printf("Cannot sign asset %v init fund txn in account %v\n", k, addr)
-				err = signErr
-				return
-			}
-			_, broadcastErr := client.BroadcastTransaction(stxn)
-			if broadcastErr != nil {
-				fmt.Printf("Cannot broadcast asset %v init fund txn in account %v\n", k, addr)
-				err = broadcastErr
-				return
-			}
-			if !cfg.Quiet {
-				fmt.Printf("Fund %d asset %d to account %s\n", assetAmt, k, addr)
+					tx, sendErr := constructTxn(addr, addr2, cfg.MaxFee, assetAmt, k, client, cfg)
+					if sendErr != nil {
+						fmt.Printf("Cannot transfer asset %v from account %v\n", k, addr)
+						err = sendErr
+						return
+					}
+
+					_, err = signAndBroadcastTransaction(accounts, addr, tx, client, cfg)
+					if err != nil {
+						_, _ = fmt.Fprintf(os.Stderr, "signing and broadcasting asset distribution failed with error %v\n", err)
```

Comment:
What's the `_, _ = ` for?

Reply:
To satisfy the linter in GoLand.  Thanks for your review!

Reply:
Alternate pattern, since this is in a function that returns error, augent the error that happened with notes about where and why it happened:
`err = fmt.Errorf("message about what's up in this function, %v", err)`
then return `err`

Reply:
Stderr works in this case because it's inside pingpong which is a pretty user oriented command line util, but I use that error pattern a lot and it works whether the thing at the top is command line or httpd or unit test.